File: ios/Classes/Wrapper/Commands/Config/LanguageCodeCommand.swift
//
// Created by Emarsys on 2021.
//
import EmarsysSDK

class LanguageCodeCommand: EmarsysCommandProtocol {
    func execute(arguments: [String : Any]?, resultCallback: @escaping ResultCallback) {
        resultCallback(["success": Emarsys.config.languageCode()])
    }
}

File: android/src/main/kotlin/com/emarsys/emarsys_sdk/command/config/LanguageCodeCommand.kt
package com.emarsys.emarsys_sdk.command.config

import com.emarsys.Emarsys
import com.emarsys.emarsys_sdk.command.EmarsysCommand
import com.emarsys.emarsys_sdk.command.ResultCallback

class LanguageCodeCommand : EmarsysCommand {
    override fun execute(parameters: Map<String, Any?>?, resultCallback: ResultCallback) {
        resultCallback.invoke(Emarsys.config.languageCode, null)
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false
        return true
    }

    override fun hashCode(): Int {
        return javaClass.hashCode()
    }
}